public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -251,29 +233,66 @@ export class PanelInspector extends PureComponent<Props, State> {
   }
 
   renderStatsTab() {
-    const { stats } = this.state;
+    const { last } = this.state;
+    const { request } = last;
+
+    if (!request) {
+      return null;
+    }
+
+    let stats: QueryResultMetaStat[] = [];
+
+    const requestTime = request.endTime ? request.endTime - request.startTime : -1;
+    const processingTime = last.timings?.dataProcessingTime || -1;
+    let dataRows = 0;
+
+    for (const frame of last.series) {
+      dataRows += frame.length;
+    }
+
+    stats.push({ title: 'Total request time', value: requestTime, unit: 'ms' });
+    stats.push({ title: 'Data processing time', value: processingTime, unit: 'ms' });
```

Comment:
hymm -- the processing time seems to time stuff that is not doing anything too heavy... need to investigate that so it is actually timing the transformations

Reply:
Yes! It's slightly more tricky tho with the transformations happening "on demand" when i.e. panel requests those